config/initializers/rack_timeout.rb

```diff
@@ -0,0 +1,2 @@
+Rack::Timeout.service_timeout = 3 # seconds
```

Comment:
3 seconds might be too generous, based on checking NewRelic, this was bigger than the average for all our requests (except when we were in an outage state a few weeks ago)

Reply:
I would be inclined to make this more like 30s or 60s. There is often a huge difference between your 50th percentile request time versus your 90th or 99th percentile slowest requests.

The way I've typically seen this done before is to have a timeout funnel: so something like 30s at the app, 45s at nginx, 60s at the load balancer.

Reply:
I bet we could do 15s? But also happy to go to 30 and we can change it later if needed

<img width="707" alt="screenshot_8_31_17__3_52_pm" src="https://user-images.githubusercontent.com/458784/29942596-771aba5a-8e64-11e7-8d3b-b26e7c6639e5.png">

Reply:
Updated to 30s